app/api/modules.py

```diff
@@ -29,7 +29,7 @@ def before_patch(self, args, kwargs, data=None):
         :param data:
         :return:
         """
-        if data.get('donation_include'):
+        if not data.get('donation_include'):
```

Comment:
???

This is wrong logic AFAIK

Reply:
After these changes, I am able to turn the donations on. But turning them off, unexpected error occurs. There was vice-versa state before these changes. What could the error possibly be?

Reply:
You are checking if the data has donation_include or not. You should check if the settings has donations enabled

Reply:
@mrsaicharan1 can you update this PR please.

Reply:
@CosmicCoder96 Yes. Will update by today. Should I be checking for donations_include in `before_patch` itself?

Reply:
>You are checking if the data has donation_include or not. You should check if the settings have donations enabled

Did you get the point he made? 
You checked if the request received contains a filed called donation_include. 
In addition to checking that, you also need to check if that field is present in the response, has its value as `true` or `false`. `data.get` only checks the existence of the field, `data['donation_include']` will give you the value of the field.

Reply:
Got it. Will update it right now.

Reply:
Also this check shouldnt be here, it should be present in the events api. If an event organizer tries to pass a value for donations, and they are not enabled in the system, an exception should be thrown.

Reply:
@CosmicCoder96 Please tell me if there's anything else which is required

Reply:
@CosmicCoder96 Can I get an update on this please?

Reply:
@iamareebjamal Can you look into this as well?